### ISSUE TITLE
jQuery.post is confusing regarding jqXHR error vs fail

### ISSUE THREAD

**Issue by medwds** [jquery/api.jquery.com#1070]
The [documentation for `jQuery.post`](http://api.jquery.com/jQuery.post/) includes under "Additional Notes" the following:

> - If a request with jQuery.post() returns an error code, it will fail silently unless the script has also called the global .ajaxError() method. Alternatively, as of jQuery 1.5, the .error() method of the jqXHR object returned by jQuery.post() is also available for error handling.

The same document notes earlier on that `jqXHR.error()` is deprecated and that `.fail()` should be used instead. Unless I'm mistaken, the above note should be updated to reflect this.

**Comment by dmethvin** [comment]
Yes it should. Actually as of 3.0 we're recommending that people [use the standard `.then` and `.catch` methods](https://jquery.com/upgrade-guide/3.0/#breaking-change-special-case-deferred-methods-removed-from-jquery-ajax).
